packaging/helm/skipper-aws/templates/external-dns/deployment.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: external-dns
+  namespace: kube-system
+  labels:
+    application: external-dns
+    version: {{ .Values.externalDnsVersion }}
+spec:
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      application: external-dns
+  template:
+    metadata:
+      labels:
+        application: external-dns
+        version: {{ .Values.externalDnsVersion }}
+{{- if .Values.kube2iam }}
+      annotations:
+        iam.amazonaws.com/role: "{{ .Values.externalDnsAwsRole }}"
+{{- end }}
+    spec:
+      dnsConfig:
+        options:
+          - name: ndots
+            value: "1"
+      priorityClassName: system-cluster-critical
+      serviceAccountName: external-dns
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:{{ .Values.externalDnsVersion }}
+        args:
+        - --source=service
+        - --source=ingress
+        - --provider=aws
+        - --registry=txt
+        - --txt-owner-id={{ .Values.region }}:{{ .Values.clusterID }}
+        - --txt-prefix={{ .Values.external_dns_ownership_prefix }}
+        - --aws-batch-change-size=100
+        resources:
+          limits:
+            cpu: 50m
+            memory: 100Mi
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 7979
+{{- if .Values.kube_aws_iam_controller }}
```

Comment:
It would be good to also have the choice of using IAM for service accounts on EKS (https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).

Happy to help with this as I know you don't use EKS. (It's just an annotation on the Service similar to the kube2iam annotation)

Reply:
It would be nice to get help here, because I have no experience with that.

Reply:
Sure. It's very similar to kube2iam (from the Kubernetes side) but rather than annotating the pod you annotate the associated service account.
e.g.
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: example
  annotations:
    eks.amazonaws.com/role-arn: arn:aws:iam::AWS_ACCOUNT_ID:role/IAM_ROLE_NAME
```

The role then needs a spcific trust relationship set up but I don't think that needs documenting in this as if they're using IAM for service accounts then they'd have already done it for any other roles.

Reply:
I added this, too